src/Microsoft.AspNetCore.Server.IntegrationTesting/Deployers/IISExpressDeployer.cs

```diff
@@ -294,5 +302,20 @@ public override void Dispose()
                 throw new Exception($"iisexpress Process {_hostProcess.Id} failed to shutdown");
             }
         }
+
+        // Transforms the web.config file to include the hostingModel="inprocess" element
+        // and adds the server type = Microsoft.AspNetServer.IIS such that Kestrel isn't added again in ServerTests
+        private void ModifyWebConfigToInProcess()
+        {
+            var webConfigFile = $"{DeploymentParameters.PublishedApplicationRootPath}/web.config";
+            var config = XDocument.Load(webConfigFile);
+            var element = config.Descendants("aspNetCore").FirstOrDefault();
+            element.SetAttributeValue("hostingModel", "inprocess");
+
+            var argsValue = element.Attribute("arguments").Value;
```

Comment:
Why add a server argument?

Reply:
You are right, this should be part of the test itself.